WorktreeMonitor crashes display when monitoring deleted worktrees

## Summary

WorktreeMonitor continues polling deleted worktree directories, causing `simple-git` to throw unhandled errors that flood stderr and destroy the terminal display, requiring a restart.

## Current Behavior

When worktrees are deleted externally (e.g., during automated PR merges that clean up feature worktrees), the WorktreeMonitor continues polling them on its 2s/10s interval. This causes:

1. `simple-git` throws: `Cannot use simple-git on a directory that does not exist`
2. Error is wrapped as `GitError: Failed to get git worktree changes`
3. Multiple error logs flood stderr, corrupting the terminal display
4. Canopy becomes unusable until restarted

**Error output:**
```
[ERROR] Git worktree changes operation failed {
  "message": "Failed to get git worktree changes",
  "name": "GitError",
  "cause": {
    "message": "Cannot use simple-git on a directory that does not exist"
  }
}
```

## Expected Behavior

When a worktree directory no longer exists:
1. The error should be caught gracefully without flooding stderr
2. The monitor should detect the directory is gone and stop itself
3. Optionally emit a `sys:worktree:remove` event to clean up UI state
4. The display should remain stable and usable

## Steps to Reproduce

1. Start Canopy in a repository with multiple worktrees
2. In another terminal, delete a worktree directory: `rm -rf /path/to/worktree`
3. Wait for the next poll cycle (2-10 seconds)
4. Observe error messages flooding the display

**Reproducibility**: Always

## Root Cause

The error originates in `getWorktreeChangesWithStats()` at [`src/utils/git.ts:285`](https://github.com/gregpriday/canopy/blob/main/src/utils/git.ts#L285):

```typescript
const git: SimpleGit = simpleGit(cwd);
```

When `cwd` doesn't exist, `simple-git` throws immediately. This propagates up through:
- `WorktreeMonitor.updateGitStatus()` (line 289)
- The polling timer callback (line 644)

While `updateGitStatus()` has a try/catch, it only handles `index.lock` errors specially. Directory-not-found errors set `mood: 'error'` and emit updates, but the monitor keeps polling and logging errors.

## Deliverables

### Code Changes

**Primary fix in `src/utils/git.ts`:**
- Add directory existence check before calling `simpleGit(cwd)`
- Throw a specific `WorktreeNotFoundError` or return a sentinel value

**Secondary fix in `src/services/monitor/WorktreeMonitor.ts`:**
- Catch directory-not-found errors in `updateGitStatus()`
- Stop the monitor and emit `sys:worktree:remove` event
- Optionally call `worktreeService.sync()` to trigger cleanup

**Alternative approach (simpler):**
- In `WorktreeMonitor.updateGitStatus()`, check `fs.existsSync(this.path)` before calling git
- If path doesn't exist, call `this.stop()` and emit removal event

### Tests

- Add test case for `WorktreeMonitor` handling deleted directories
- Add test case for `getWorktreeChangesWithStats` with non-existent path

### Documentation

None required - internal resilience improvement.

## Technical Specifications

**Footprint:** 
- `src/utils/git.ts` - Add existence check
- `src/services/monitor/WorktreeMonitor.ts` - Handle graceful cleanup

**Performance Considerations:**
- `fs.existsSync()` is synchronous but very fast for this use case
- Only called once per poll cycle (2-10s), negligible overhead

## Dependencies

None

## Tasks

- [ ] Add directory existence check in [`src/utils/git.ts:getWorktreeChangesWithStats()`](https://github.com/gregpriday/canopy/blob/main/src/utils/git.ts#L285)
- [ ] Handle "directory not found" error in [`WorktreeMonitor.updateGitStatus()`](https://github.com/gregpriday/canopy/blob/main/src/services/monitor/WorktreeMonitor.ts#L273-L440)
- [ ] Stop monitor and emit removal event when worktree directory is deleted
- [ ] Add test coverage for deleted worktree scenario
- [ ] Verify fix doesn't interfere with normal worktree removal via `WorktreeService.sync()`

## Acceptance Criteria

- [ ] Deleting a worktree directory externally does not flood stderr with errors
- [ ] Canopy display remains stable when worktrees are deleted
- [ ] Monitor stops polling after detecting directory deletion
- [ ] UI removes the deleted worktree card (via existing event handling)
- [ ] No regression in normal worktree add/remove workflow

## Edge Cases & Risks

- **Race condition**: Directory could be deleted between existence check and git operation. Use try/catch as secondary protection.
- **Temporary unavailability**: Network drives or mounts might temporarily disappear. Consider a retry count before permanent removal.
- **Multiple rapid deletions**: During batch PR merges, many worktrees may be deleted quickly. Ensure event emission doesn't overwhelm the UI.

--- a/src/services/monitor/WorktreeMonitor.ts
+++ b/src/services/monitor/WorktreeMonitor.ts
@@ -2,6 +2,7 @@ import { EventEmitter } from 'events';
 import { createHash } from 'crypto';
 import type { Worktree, WorktreeChanges, WorktreeMood } from '../../types/index.js';
 import { getWorktreeChangesWithStats, invalidateGitStatusCache } from '../../utils/git.js';
+import { WorktreeRemovedError } from '../../utils/errorTypes.js';
 import { generateWorktreeSummary } from '../ai/worktree.js';
 import { categorizeWorktree } from '../../utils/worktreeMood.js';
 import { logWarn, logError, logInfo, logDebug } from '../../utils/logger.js';
@@ -424,6 +425,18 @@ export class WorktreeMonitor extends EventEmitter {
       }
 
     } catch (error) {
+      // Handle worktree deletion: stop monitoring and emit removal event
+      if (error instanceof WorktreeRemovedError) {
+        logInfo('Worktree directory deleted, stopping monitor', { id: this.id, path: this.path });
+
+        // Emit removal event before stopping so UI can clean up
+        events.emit('sys:worktree:remove', { worktreeId: this.id });
+
+        // Stop monitoring this worktree (clears timers, removes listeners)
+        void this.stop();
+        return;
+      }
+
       // Handle index.lock collision gracefully (don't set mood to error)
       const errorMessage = (error as Error).message || '';
       if (errorMessage.includes('index.lock')) {

--- a/src/utils/errorTypes.ts
+++ b/src/utils/errorTypes.ts
@@ -29,6 +29,16 @@ export class GitError extends CanopyError {
 }
 
 /**
+ * Worktree directory no longer exists (deleted externally).
+ * Used to signal that a worktree monitor should stop polling and clean up.
+ */
+export class WorktreeRemovedError extends GitError {
+  constructor(path: string, cause?: Error) {
+    super('Worktree directory no longer exists', { path }, cause);
+  }
+}
+
+/**
  * File system operation failed (permission denied, file not found, read error)
  */
 export class FileSystemError extends CanopyError {

--- a/src/utils/git.ts
+++ b/src/utils/git.ts
@@ -2,7 +2,7 @@ import { dirname, resolve } from 'path';
 import { realpathSync, promises as fs } from 'fs';
 import simpleGit, { SimpleGit, StatusResult } from 'simple-git';
 import type { FileChangeDetail, GitStatus, WorktreeChanges } from '../types/index.js';
-import { GitError } from './errorTypes.js';
+import { GitError, WorktreeRemovedError } from './errorTypes.js';
 import { logWarn, logError } from './logger.js';
 import { Cache } from './cache.js';
 import { perfMonitor } from './perfMetrics.js';
@@ -280,6 +280,20 @@ export async function getWorktreeChangesWithStats(
   // PERF: Limit the number of files we calculate stats for to prevent CPU hang
   // on massive changesets (e.g., accidental package-lock.json deletion in monorepo)
   const MAX_FILES_FOR_NUMSTAT = 100;
+
+  // Check if directory exists before calling simpleGit.
+  // simple-git throws immediately if cwd doesn't exist, which can flood stderr
+  // when a worktree is deleted externally while being monitored.
+  try {
+    await fs.access(cwd);
+  } catch (accessError) {
+    const nodeError = accessError as NodeJS.ErrnoException;
+    if (nodeError.code === 'ENOENT') {
+      throw new WorktreeRemovedError(cwd, nodeError);
+    }
+    // Re-throw other access errors (e.g., permissions)
+    throw accessError;
+  }
 
   try {
     const git: SimpleGit = simpleGit(cwd);

--- a/tests/services/WorktreeMonitor.test.ts
+++ b/tests/services/WorktreeMonitor.test.ts
@@ -4,6 +4,7 @@ import type { Worktree } from '../../src/types/index.js';
 import { events } from '../../src/services/events.js';
 import * as gitStatus from '../../src/utils/git.js';
 import * as worktreeMood from '../../src/utils/worktreeMood.js';
+import { WorktreeRemovedError } from '../../src/utils/errorTypes.js';
 
 // Mock the git utilities
 vi.mock('../../src/utils/git.js', () => ({
@@ -516,6 +517,76 @@ describe('WorktreeMonitor - Atomic State Updates', () => {
       expect(monitor.getState().mood).toBe('error');
 
       await monitor.stop();
+    });
+
+    it('stops monitor and emits removal event when worktree is deleted', async () => {
+      vi.mocked(gitStatus.getWorktreeChangesWithStats)
+        .mockResolvedValueOnce({
+          worktreeId: baseWorktree.id,
+          rootPath: baseWorktree.path,
+          changes: [],
+          changedFileCount: 0,
+          lastUpdated: Date.now(),
+        })
+        .mockRejectedValueOnce(new WorktreeRemovedError(baseWorktree.path));
+
+      const monitor = new WorktreeMonitor(baseWorktree);
+      await monitor.start();
+
+      // Track removal events
+      const removedWorktrees: string[] = [];
+      const handler = (payload: { worktreeId: string }) => {
+        removedWorktrees.push(payload.worktreeId);
+      };
+      events.on('sys:worktree:remove', handler);
+
+      const stopSpy = vi.spyOn(monitor, 'stop');
+
+      // Trigger update that will fail with WorktreeRemovedError
+      await (monitor as any).updateGitStatus();
+
+      // Should emit removal event
+      expect(removedWorktrees).toContain(baseWorktree.id);
+
+      // Should stop the monitor
+      expect(stopSpy).toHaveBeenCalled();
+
+      events.off('sys:worktree:remove', handler);
+    });
+
+    it('stops polling when worktree is deleted during monitoring', async () => {
+      let callCount = 0;
+      vi.mocked(gitStatus.getWorktreeChangesWithStats).mockImplementation(async () => {
+        callCount++;
+        if (callCount === 1) {
+          // First call succeeds (initial state)
+          return {
+            worktreeId: baseWorktree.id,
+            rootPath: baseWorktree.path,
+            changes: [],
+            changedFileCount: 0,
+            lastUpdated: Date.now(),
+          };
+        }
+        // Subsequent calls fail with directory removed
+        throw new WorktreeRemovedError(baseWorktree.path);
+      });
+
+      const monitor = new WorktreeMonitor(baseWorktree);
+      await monitor.start();
+
+      // Advance timer to trigger polling
+      await vi.advanceTimersByTimeAsync(2000);
+
+      // The monitor should have stopped after detecting removal
+      // This means no further polling calls should happen
+      const callCountAfterRemoval = callCount;
+
+      // Advance more time to verify no more calls
+      await vi.advanceTimersByTimeAsync(10000);
+
+      // Call count should not increase after removal was detected
+      expect(callCount).toBe(callCountAfterRemoval);
     });
   });
 

--- a/tests/utils/git.test.ts
+++ b/tests/utils/git.test.ts
@@ -1,6 +1,7 @@
 import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
 import { getGitStatusCached, clearGitStatusCache, stopGitStatusCacheCleanup, getWorktreeChangesWithStats } from '../../src/utils/git.js';
 import type { GitStatus } from '../../src/types/index.js';
+import { WorktreeRemovedError } from '../../src/utils/errorTypes.js';
 
 // Mock simple-git
 vi.mock('simple-git', () => ({
@@ -27,6 +28,7 @@ vi.mock('fs', async () => {
 		promises: {
 			readFile: vi.fn(),
 			stat: vi.fn(),
+			access: vi.fn(), // Add access mock for directory existence check
 		},
 	};
 });
@@ -330,6 +332,91 @@ describe('git.ts', () => {
 			expect(change.status).toBe('untracked');
 			expect(change.insertions).toBeNull(); // Binary files return null
 			expect(change.deletions).toBeNull();
+		});
+	});
+
+	describe('getWorktreeChangesWithStats - deleted directory handling', () => {
+		let mockAccess: any;
+		let mockStat: any;
+
+		beforeEach(async () => {
+			clearGitStatusCache();
+			// Get the mocked fs module
+			const fs = await import('fs');
+			mockAccess = fs.promises.access as any;
+			mockStat = fs.promises.stat as any;
+
+			// Reset mocks
+			vi.mocked(mockAccess).mockReset();
+			vi.mocked(mockStat).mockReset();
+			vi.mocked(mockStat).mockResolvedValue({ mtimeMs: Date.now() } as any);
+		});
+
+		it('throws WorktreeRemovedError when directory does not exist', async () => {
+			// Mock fs.access to fail with ENOENT
+			const enoentError = new Error('ENOENT: no such file or directory');
+			(enoentError as NodeJS.ErrnoException).code = 'ENOENT';
+			vi.mocked(mockAccess).mockRejectedValue(enoentError);
+
+			await expect(getWorktreeChangesWithStats('/deleted/worktree', true))
+				.rejects
+				.toThrow(WorktreeRemovedError);
+		});
+
+		it('includes path in WorktreeRemovedError context', async () => {
+			const enoentError = new Error('ENOENT: no such file or directory');
+			(enoentError as NodeJS.ErrnoException).code = 'ENOENT';
+			vi.mocked(mockAccess).mockRejectedValue(enoentError);
+
+			try {
+				await getWorktreeChangesWithStats('/deleted/worktree', true);
+				expect.fail('Should have thrown');
+			} catch (error) {
+				expect(error).toBeInstanceOf(WorktreeRemovedError);
+				expect((error as WorktreeRemovedError).context?.path).toBe('/deleted/worktree');
+			}
+		});
+
+		it('re-throws non-ENOENT errors without wrapping', async () => {
+			// Mock fs.access to fail with permission error
+			const epermError = new Error('EPERM: operation not permitted');
+			(epermError as NodeJS.ErrnoException).code = 'EPERM';
+			vi.mocked(mockAccess).mockRejectedValue(epermError);
+
+			await expect(getWorktreeChangesWithStats('/protected/worktree', true))
+				.rejects
+				.toThrow('EPERM: operation not permitted');
+
+			// Should NOT be a WorktreeRemovedError
+			try {
+				await getWorktreeChangesWithStats('/protected/worktree', true);
+			} catch (error) {
+				expect(error).not.toBeInstanceOf(WorktreeRemovedError);
+			}
+		});
+
+		it('proceeds normally when directory exists', async () => {
+			const simpleGit = await import('simple-git');
+			vi.mocked(simpleGit.default).mockReturnValue({
+				status: vi.fn().mockResolvedValue({
+					modified: ['file.ts'],
+					created: [],
+					deleted: [],
+					renamed: [],
+					not_added: [],
+					conflicted: [],
+				}),
+				revparse: vi.fn().mockResolvedValue('/existing/repo'),
+				diff: vi.fn().mockResolvedValue('10\t5\tfile.ts'),
+			} as any);
+
+			// Mock fs.access to succeed (directory exists)
+			vi.mocked(mockAccess).mockResolvedValue(undefined);
+
+			const result = await getWorktreeChangesWithStats('/existing/repo', true);
+
+			expect(result.changes).toHaveLength(1);
+			expect(result.changes[0].status).toBe('modified');
 		});
 	});
 });

